Pages/FoodCustomer/Cart.cshtml.cs

```diff
@@ -91,9 +93,19 @@ public async Task<IActionResult> OnPostRemoveFromCartAsync(int id)
         }
 
 
-        public IActionResult OnPostReserve()
+        public async Task <IActionResult> OnPostReserve()
         {
             OnGet();
+            foreach (var item in cart)
+            {
+                var food = item.Food;
+                food.Quantity = item.Food.Quantity - item.Quantity;
```

Comment:
This could have been commented to make it clear what is happening. It is quite confusing.

Reply:
Updating food's quantity in DB after customer reserved it